### DISCUSSION TITLE
Request for WinkHaus WindowHandler

### DISCUSSION THREAD

**Discussion by SmokyMcPott** [mak-gitdev/HA_enoceanmqtt#63] [Request] [answered]
Salut mak,
first of all, I would like to express my sincere thanks for your work!
I am planning to automate the opening and closing of the windows in my house. This is done through the motor from [WinkHaus](https://www.winkhaus.com/en/fbt/activpilot), which communicates via Enocean.

Here, https://shop.wibutler.com/sicherheit/fensterantriebe/74/winkhaus-parallelabstellfenster-antrieb-hf.md.padm.01-grau/silber, it is stated that the EEP should be F6-02-01. However, according to the [official website](https://www.enocean-alliance.org/products/eeps/), this cannot be correct.

How can I find out the correct EEP?
And could you add this device?

I would be very grateful for a response and thank you in advance!
À bientôt,
SmokyMcPott

**Comment by mak-gitdev** [comment]
Hi @SmokyMcPott,

> However, according to the [official website](https://www.enocean-alliance.org/products/eeps/), this cannot be correct.

I suppose you think it isn't correct because F6-02-01 is described as `Light and Blind Control` while there is a dedicated EEP for Window Handle (F6-10-00), right ?

I think manufacturers can use the EEP that better suits there needs. So maybe in this case, the guys at WinkHaus preferred to use F6-02-01 instead of F6-10-00.

I think you should try with the indicated EEP to check whether it works or not.

**Comment by cubertt** [comment, marked as answer]
I think the difference is that the WinkHaus WindowHandler is an active open close device (it opens the window) and F6-10-00 is more a sensor (is the window open?).

So its more comparable to a cover. Perhaps you could teach in a virtual switch? (if the suggestio of mak-gitdev is not working)

**Comment by mak-gitdev** [comment]
@cubertt is absolutely right !

I missed that the direction was from HA to the WinkHaus.
I thought the WinkHaus was a sensor.
So the virtual switch is definitely the way to go.

Thanks @cubertt :+1:

**Comment by SmokyMcPott** [comment]
Thank you both for the responses.

> I suppose you think it isn't correct because F6-02-01 is described as `Light and Blind Control` while there is a dedicated EEP for Window Handle (F6-10-00), right ?
> 
Yes, that was my idea. I was confused that I could read the communication of an Eltako switch F2(Z)/F4T55E via enoceanmqtt.log, but the WinkHaus handler doesn't send any radio signals even in learning mode.
That's why I thought it might be due to the wrong EEP.

You recommend that I should integrate a virtual switch. After reading your messages, I have looked into it a bit, but I'm not yet confident with the topic.

If I understand correctly, the WinkHaus handler doesn't send signals because it is not a sensor and can only receive commands, right? Therefore, I need the virtual switch to send the commands to the WinkHaus handler?
Please correct me if I'm mistaken.

**Comment by mak-gitdev** [comment]
@SmokyMcPott,

> If I understand correctly, the WinkHaus handler doesn't send signals because it is not a sensor and can only receive commands, right? Therefore, I need the virtual switch to send the commands to the WinkHaus handler?

Yes that's my understanding.

**Comment by SmokyMcPott** [comment]
Okay, it took a bit longer for the theoretical considerations to work in practice. But now it's working!! 

I had trouble in two areas:
1. The [wiki](https://github.com/mak-gitdev/HA_enoceanmqtt/wiki/Usage#1--pairing-your-device) referred to the base ID, but I wasn't exactly sure what that meant. Is it referring to the ID of my USB-300 dongle, right? At least, that's what worked.
2. The WinkHaus handler's [manual](https://www.manualslib.com/manual/2141042/Winkhaus-Hf-Md-Padm-01.html?page=27#manual) describes that the device needs to be paired by pressing the switch three times. Of course, this also needs to be simulated with the virtual switch to complete the pairing process.

I'm marking @cubertt  response as the solution to this problem.

Thanks a lot for your help. And thanks again @mak-gitdev for the addon!
